pandas/io/pytables.py

```diff
@@ -303,6 +303,9 @@ def read_hdf(path_or_buf, key=None, **kwargs):
 
         """
 
+    if kwargs.get('mode') == 'w':
```

Comment:
Fixed. Now instead of checking for invalid modes, I check only for valid modes and raise a ValueError if any invalid modes are passed.